pkg/apis/provisioning/v1alpha3/provisioner_validation.go

```diff
@@ -101,6 +102,13 @@ func (c *Cluster) validate() (errs *apis.FieldError) {
 	}
 	if len(c.Endpoint) == 0 {
 		errs = errs.Also(apis.ErrMissingField("endpoint"))
+	} else {
+		endpoint, err := url.Parse(c.Endpoint)
+		// url.Parse() will accept a lot of input without error; make
+		// sure it's a real URL
+		if err != nil || !endpoint.IsAbs() || endpoint.Hostname() == "" {
+			errs = errs.Also(apis.ErrInvalidValue(c.Endpoint, "endpoint"))
```

Comment:
Can you write in a helpful message about what it should be (e.g. a valid uri). IIRC, you can put this message into the .Also()

Reply:
done